### ISSUE TITLE
the use of separate trained networks for each camera

### ISSUE THREAD

**Issue by MaudvdBogaart** [lambdaloop/anipose#23]
Hello,
I looked at the data set; ‘anipose-shared: hand-demo’ of the anipose tutorial (great, thank you for this!) and I noticed one network was trained on all views/cameras. I was wondering if it is also possible to use separate trained networks for each camera?
Best regards, 
Maud

**Comment by lambdaloop** [comment]
Hi @MaudvdBogaart ,
At this moment, it's not possible to use separate networks for each camera. It's possible in principle, but from our brief experiments we didn't find any negative effects from just training one network on all the data. In fact, if anything, there was a slight positive bonus as the network trained on multiple views could generate slightly better. 

If this feature is crucial to you, let me know and I could look into implementing it.

**Comment by MaudvdBogaart** [comment]
@lambdaloop , thanks for the quick response. 
Okay perfect, good to know. 
It is not a problem to just training one network on multiple views, so the feature to use seperate networks is not essential for me, thank you.

**Comment by MaudvdBogaart** [comment]
Hi @lambdaloop 
I previously asked if it is possible to use separate trained networks for each camera. At first attempt it seemed not crucial to do this, but when comparing the labeling (in DeepLabCut) when training per camera and on multiple cameras at once, we concluded that the labelling was substantial better when trained per camera.
The cameras are positioned in front but also at the back of the subject, so we could imagine that our setup (8 cameras surrounding the subject) would be the reason for this substantial better labeling when training per camera (viz. Markers on the back are never visible in the front cameras for example and vice versa)
Would it be an option to work on an implementation of the possibility to use seperate training networks for each camera in anipose to retrieve 3D coordinates of the labels? 
Best regards,
Maud van den Bogaart

**Comment by SjoerdBruijn** [comment]
I have figured this out today. Will do a pull request soon.

**Comment by SjoerdBruijn** [comment]
@lambdaloop I think this can be closed since pull request Update triangulate.py #55 was merged? Or do you want to keep it open as potenntial future issue to work on improvements so that data from different models doesnt have to be put in the same folder?

**Comment by lambdaloop** [comment]
Yeah, I'll leave it open for now. As I see it, supporting separate networks per camera would be best if you could configure which network goes to which camera in the `config.toml` file and have anipose handle the rest
